### PR TITLE
Inline Patterns feature and pattern rendering refactor-ish

### PR DESCRIPTION
**Pull request by SamsTheNerd**
See https://github.com/FallingColors/HexMod/pull/708 for details. This is (mostly) the same but cleaned up into fewer commits.

### CODE REVIEW

File: Common/src/main/java/at/petrak/hexcasting/api/casting/iota/EntityIota.java [hunk 1]
```diff
@@ -73,6 +94,7 @@ public Component display(Tag tag) {
                 return Component.translatable("hexcasting.spelldata.entity.whoknows");
             }
             var nameJson = ctag.getString("name");
+//            return Component.literal(nameJson);
```
Comment (vgskye) [inline Common/src/main/java/at/petrak/hexcasting/api/casting/iota/EntityIota.java:97]: vestigial debug statement?

File: Common/src/main/java/at/petrak/hexcasting/api/casting/iota/PatternIota.java [hunk 2]
```diff
@@ -172,7 +179,7 @@ public static Component display(HexPattern pat) {
             bob.append(sig);
         }
         return Component.translatable("hexcasting.tooltip.pattern_iota",
-                Component.literal(bob.toString()).withStyle(ChatFormatting.WHITE))
-            .withStyle(ChatFormatting.GOLD);
+                        Component.literal(bob.toString()).withStyle(ChatFormatting.WHITE))
+                .withStyle(ChatFormatting.GOLD);
```
Comment (vgskye) [inline Common/src/main/java/at/petrak/hexcasting/api/casting/iota/PatternIota.java:183]: indent change?

File: Common/src/main/java/at/petrak/hexcasting/client/entity/WallScrollRenderer.java [hunk 3]
```diff
@@ -64,7 +58,9 @@ public void render(EntityWallScroll wallScroll, float yaw, float partialTicks, P
             var mat = last.pose();
             var norm = last.normal();
 
-            var verts = bufSource.getBuffer(RenderType.entityCutout(this.getTextureLocation(wallScroll)));
+            RenderType layer = RenderType.entityCutout(this.getTextureLocation(wallScroll));
+
+            var verts = bufSource.getBuffer(layer);
```
Comment (vgskye) [inline Common/src/main/java/at/petrak/hexcasting/client/entity/WallScrollRenderer.java:63]: why did layer get extracted into a variable here?

File: Common/src/main/java/at/petrak/hexcasting/client/entity/WallScrollRenderer.java [hunk 4]
```diff
@@ -163,146 +125,12 @@ public ResourceLocation getTextureLocation(EntityWallScroll wallScroll) {
...
-                var theta = (i - j) / (float) fracOfCircle * Mth.TWO_PI;
-                var rx = Mth.cos(theta) * radius + point.x;
-                var ry = Mth.sin(theta) * radius + point.y;
-                vertexCol(mat, normal, light, verts, color, new Vec2(rx, ry));
-            }
-        }
+                .color(0xffffffff)
+                .uv(u, v).overlayCoords(OverlayTexture.NO_OVERLAY).uv2(light)
+                .normal(normal, nx, ny, nz)
+                .endVertex();
```
Comment (vgskye) [inline Common/src/main/java/at/petrak/hexcasting/client/entity/WallScrollRenderer.java:134]: indent changes(again)

File: Common/src/main/java/at/petrak/hexcasting/client/render/PatternRenderer.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            vcHelper.vcEndDrawer(vc);
+        }
+
+        return true;
+    }
+
+    // TODO did we want to un-hardcode this for accessibility reasons ?
+    public static boolean shouldDoStrokeGradient(){
+        return Screen.hasControlDown();
+    }
```
Comment (vgskye) [inline Common/src/main/java/at/petrak/hexcasting/client/render/PatternRenderer.java:148]: maybe switch this over to sprint bind?

File: Common/src/main/java/at/petrak/hexcasting/interop/inline/HexPatternMatcher.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        return PATTERN_PATTERN_REGEX;
+    }
+
+    @Override
+    @NotNull
+    public Tuple<InlineMatch, Integer> getMatchAndGroup(MatchResult regexMatch, MatchContext ctx) {
+        String escaped = regexMatch.group(1);
+        String dirString = regexMatch.group(2).toLowerCase().strip().replace("_", "");
+        String sizeModString = regexMatch.group(3);
+        String angleSigs = regexMatch.group(4);
```
Comment (vgskye) [inline Common/src/main/java/at/petrak/hexcasting/interop/inline/HexPatternMatcher.java:43]: why have named capturing groups and then index em with numbers?

**Comment by gamma-delta** [comment]
Sure LGTM I guess

**Pull request merged**